File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int curr = nums[0];
        int counter = 0;
        for(int i = 0 ; i < nums.size() ; i++){
            if(nums[i] == curr){
                counter++;
            }
            else{
                counter--;
                if(counter == 0){
                    curr = nums[i];
                    counter = 1;
                }
            }
        }
        return curr;
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l , r = 0 , len(nums)-1
        sol = len(nums)
        while l <= r : 
            m = (l + r) // 2
            if nums[m] == target:
                return m
            elif nums[m] > target:
                r = m - 1
                sol = m 
            else:
                l = m + 1
        return sol